### PR TITLE
Support meta-types in the TF backend

### PR DESCRIPTION
**Pull request by alexsu52**
- Introduced meta-type in the NNCFNode
- Using meta-types in the TF backend instead of framework types
- Migrated quantization and sparsities algorithms to using NNCFGraph.

### CODE REVIEW

File: nncf/common/graph/graph.py [hunk 1]
```diff
@@ -41,6 +42,10 @@ def node_name(self) -> str:
     def node_type(self) -> str:
         return self.data.get(NNCFGraph.NODE_TYPE_ATTR)
 
+    @property
+    def metatype(self) -> Type[OperatorMetatype]:
+        return self.data.get(NNCFGraph.METATYPE_ATTR)
```
Comment (vshampor) [inline nncf/common/graph/graph.py:47]: How should this be different from `node_type`?

File: beta/nncf/tensorflow/graph/converter.py [hunk 2]
```diff
@@ -69,10 +45,17 @@ def _get_layer_dtype(layer):
...
+    try:
+        return model.get_layer(layer_name)
+    except ValueError:
+        for layer in model.submodules:
+            if not isinstance(layer, tf.keras.layers.Layer):
+                continue
+            if layer.name == layer_name:
+                return layer
+
+    raise ValueError('No such layer: ' + layer_name + '.')
```
Comment (andrey-churkin) [inline beta/nncf/tensorflow/graph/converter.py:58]:
```suggestion
    raise ValueError(f'No such layer: {layer_name}.')
```

File: beta/nncf/tensorflow/graph/metatypes/keras_layers.py [hunk 3]
```diff
@@ -19,40 +19,51 @@
...
-    name = 'noop'
-
-
-#
-# TensorFlow Keras Layer Metatypes Section
-#
 class TFLayerMetatype(OperatorMetatype):
     keras_layer_names = []  # type: List[str]
-    subtypes = []  #type: List[Type['OperatorMetatype']]
+    subtypes = []  #type: List[Type[OperatorMetatype]]
```
Comment (andrey-churkin) [inline beta/nncf/tensorflow/graph/metatypes/keras_layers.py:31]:
```suggestion
    subtypes = []  # type: List[Type[OperatorMetatype]]
```

**Review by andrey-churkin — CHANGES_REQUESTED** [review]

File: nncf/common/graph/graph.py (hunk 1 above)
Comment (alexsu52) [reply nncf/common/graph/graph.py:47]: Node_type is framework operator type, but metatype by definition is a type for grouping framework operators based on their semantic meaning. In the general case, we can not determine framework operator type by metatype.

File: beta/nncf/tensorflow/graph/converter.py (hunk 2 above)
Comment (alexsu52) [reply beta/nncf/tensorflow/graph/converter.py:58]: done

File: beta/nncf/tensorflow/graph/metatypes/keras_layers.py (hunk 3 above)
Comment (alexsu52) [reply beta/nncf/tensorflow/graph/metatypes/keras_layers.py:31]: done

**Comment by alexsu52** [comment]
Jenkins please retry a build

**Review by andrey-churkin — APPROVED** [review]

**Pull request merged**